pkg/ebpf/extension.go

```diff
@@ -0,0 +1,96 @@
+package ebpf
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/aquasecurity/tracee/common/errfmt"
+)
+
+// InitPhase represents different phases of Tracee initialization
+// Extensions should implement a switch case in their Init method to handle different phases
+type InitPhase int
+
+const (
+	// Early initialization phases
+	InitPhaseStart         InitPhase = iota // At the start of Tracee Init
+	InitPhaseCGroups                        // After cgroups filesystems initialization
+	InitPhaseContainers                     // After containers initialization
+	InitPhaseBPFProbes                      // After eBPF probes initialization
+	InitPhaseKernelSymbols                  // After kernel symbols initialization
+	InitPhaseBPFPrograms                    // After eBPF programs and maps initialization
+	InitPhaseComplete                       // At the end of Tracee Init
+)
+
+// String returns the string representation of the InitPhase
+func (p InitPhase) String() string {
+	switch p {
+	case InitPhaseStart:
+		return "InitPhaseStart"
+	case InitPhaseCGroups:
+		return "InitPhaseCGroups"
+	case InitPhaseContainers:
+		return "InitPhaseContainers"
+	case InitPhaseBPFProbes:
+		return "InitPhaseBPFProbes"
+	case InitPhaseKernelSymbols:
+		return "InitPhaseKernelSymbols"
+	case InitPhaseBPFPrograms:
+		return "InitPhaseBPFPrograms"
+	case InitPhaseComplete:
+		return "InitPhaseComplete"
+	default:
+		return fmt.Sprintf("UNKNOWN_PHASE(%d)", int(p))
+	}
+}
+
+// Extension interface for Tracee
+type Extension interface {
+	// Initialize the extension with a specific phase
+	Init(ctx context.Context, t *Tracee, phase InitPhase) error
+	// Run the extension
+	Run(ctx context.Context, t *Tracee) error
+	// Close the extension
+	Close(t *Tracee) error
+}
+
+var extensionRegistry []Extension
+
+func RegisterExtension(ext Extension) {
+	extensionRegistry = append(extensionRegistry, ext)
+}
+
+// loadExtensions loads all registered extensions from the global registry.
+func (t *Tracee) loadExtensions() {
+	t.extensions = append([]Extension(nil), extensionRegistry...)
+}
```

Comment:
Do we really need to load the extenstion?
Isn't it enough to keep them in extensionRegistry defined here and add a getter function instead?

I would also consider making Extensions a struct and all the methods of this file can be of the Extensions struct and not of Tracee struct

Reply:
In this case we need to load the extensions since the extension were created before Tracee struct creation - since the extensions uses `init()`... but I've change it... WDYT?

Agree that is more organize to make those methods of the Extensions struct... the only one which need to be package level is `RegisterExtension` since the extension `init()` call it before Tracee struct initialization. Using this approach I needed to still use the field extension in the Tracee... WDYT of this approach?